src/pc/1.17/ChunkColumn.js

```diff
@@ -0,0 +1,266 @@
+      return sectionCoord - bottomSectionCoord
+    }
+
+    getBiomeIndex (pos) {
+      const i = (pos.x >> 2) & 3
+      const j = (pos.y >> 2) - (this.minWorldHeight >> 2)
+      const k = (pos.z >> 2) & 3
+      return j << 4 | k << 2 | i
+    }
+
+    getBottomSectionCoord () {
+      return getSectionCoord(this.minWorldHeight)
+    }
+
+    getTopSectionCoord () {
+      return getSectionCoord(this.maxWorldHeight - 1) + 1
+    }
+
+    isBlockPosValid (pos) {
+      return pos.y >= this.minWorldHeight && pos.y < this.maxWorldHeight &&
+        pos.x >= 0 && pos.x < constants.SECTION_WIDTH &&
+        pos.z >= 0 && pos.z < constants.SECTION_WIDTH
+    }
+
+    // Block State retrieval methods
+    getBlockStateId (pos) {
+      if (this.isBlockPosValid(pos)) {
+        const section = this.sections[this.getSectionIndex(pos.y)]
+        return section ? section.getBlock(toSectionPos(pos)) : 0
+      }
+      return 0
+    }
+
+    setBlockStateId (pos, stateId) {
+      if (!this.isBlockPosValid(pos)) {
+        return
+      }
+
+      const sectionIndex = this.getSectionIndex(pos.y)
+      let section = this.sections[sectionIndex]
+
+      if (!section) {
+        // if it's air, do not create a new section for it
+        if (stateId === 0) {
+          return
+        }
+        section = new ChunkSection()
+        this.sectionMask |= 1 << sectionIndex
+        this.sections[sectionIndex] = section
+      }
+      section.setBlock(toSectionPos(pos), stateId)
+    }
+
+    // Biome methods
+    getBiome (pos) {
+      if (this.isBlockPosValid(pos)) {
+        return this.biomes[this.getBiomeIndex(pos)]
+      }
+      return 0
+    }
+
+    setBiome (pos, biome) {
+      if (this.isBlockPosValid(pos)) {
+        this.biomes[this.getBiomeIndex(pos)] = biome
+      }
+    }
+
+    // Convenience methods returning Block object
+    getBlock (pos) {
+      const biome = this.getBiome(pos)
+      const stateId = this.getBlockStateId(pos)
+
+      return Block.fromStateId(stateId, biome)
+    }
+
+    setBlock (pos, block) {
+      if (typeof block.stateId !== 'undefined') {
+        this.setBlockStateId(pos, block.stateId)
+      }
+      if (typeof block.biome !== 'undefined') {
+        this.setBiome(pos, block.biome.id)
+      }
+    }
+
+    // Legacy conversion methods accepting block type + metadata
+    getBlockType (pos) {
+      const blockStateId = this.getBlockStateId(pos)
+      return mcData.blocksByStateId[blockStateId].id
+    }
+
+    getBlockData (pos) {
+      const blockStateId = this.getBlockStateId(pos)
+      return mcData.blocksByStateId[blockStateId].metadata
+    }
+
+    setBlockType (pos, id) {
+      this.setBlockStateId(pos, mcData.blocks[id].minStateId)
+    }
+
+    setBlockData (pos, data) {
+      this.setBlockStateId(pos, mcData.blocksByStateId[this.getBlockStateId(pos)].minStateId + data)
+    }
+
+    // Skylight data has been removed in 1.17, now it is computed client-side automatically
```

Comment:
pchunk is used both server and client side
Even if it's not serialized, we still need to store it, if it's something that is needed for rendering

Reply:
I mean, realistically, we have no way of computing it without writing our own lighting engine. It's also never stored on server too, lightning is completely separate from chunk persistence and serialization now.
even though I could make it store lighting data, what would be the source of it?

Reply:
There are ongoing projects to write a prismarine implementation of the lighting engine. And the world generators for flying-squid do generate (simple) light data. Independently of whether or not it is used somewhere, storing light is supported by other chunk version so it would be a breaking change to not support it here.

Reply:
Alright, brought implementation back